<html><head>
<title>rna</title>
<style type="text/css">
A.nounder:link {text-decoration: none}
A.nounder:visited {text-decoration: none}
A.nounder:active {text-decoration: none}
A.nounder:hover {text-decoration: underline; color: red;}
</style>
</head><body>
<font size="-1">
<a href="../../chimera/helpdir/UsersGuide/framecommand.html" target="_top">Chimera Commands Index</a>
</font>
<a name="top"></a>
<h3><a href="../../chimera/helpdir/UsersGuide/midas/usageconventions.html">Usage</a>:
<BR><b>rna</b> <i>operation</i> <i>arguments</i></h3>
<p>
The <b>rna</b> command builds models of RNA or DNA molecules from 
secondary structure (base pairing) information.  Nucleotides are layed
out in a schematic way for stems and loops in 3 dimensions. This
capability was used to make the 9000 nucleotide HIV virus RNA and DNA
models used in an <a href="http://youtu.be/i-__ufO9LYg">HIV virus animation</a>.
The <i>operation</i> can be:
<ul>
<li><a href="#duplex"><b>duplex</b></a> - create double helical DNA models following a given path.
<li><a href="#minimizeBackbone"><b>minimizeBackbone</b></a> - energy minimize RNA/DNA backbones leaving bases fixed.
<li><a href="#model"><b>model</b></a> - create single strand RNA models from secondary structure.
<li><a href="#path"><b>path</b></a> - create path of markers representing stems and loops.
<li><a href="#smoothPath"><b>smoothPath</b></a> - create a smoother version of a path.
</ul>
<p>
Operation keywords and their sub-keywords described below
can be truncated to unique strings, and their case does not matter.
Synonyms for true: True, 1.  Synonyms for false: False, 0.
A vertical bar "|" designates mutually exclusive options,
and default settings are indicated with <b>bold</b>.
</p>

<a href="#top" class="nounder">&bull;</a>
  <a name="duplex"><b>rna duplex</b></a> &nbsp;<i>path</i>&nbsp; &nbsp;<i>sequence</i>&nbsp;
  [&nbsp;<b>startSequence</b>&nbsp;<i>i</i>&nbsp;]
  [&nbsp;<b>type</b>&nbsp;<b>DNA</b>&nbsp;|&nbsp;RNADNA&nbsp;]
<br>
<blockquote>
Creates an atomic model of double helical DNA or RNA-DNA hybrid with
a specified sequence on one strand, the complementary sequence on the other
strand and following a specified marker path where each marker represents
one basepair.  The <b>path</b> argument specifies a sequence of Volume
Tracer markers such as that produced by the rna command path or smoothPath
operations or atoms which give the center for each basepair.  The
<b>sequence</b> argument can be a string containing upper case letters
AGCTU or a path to a FASTA file containing the desired sequence.  The
optional <b>startSequence</b> argument (default 1) specifies where to start in
the specified sequence.  The optional <b>type</b> argument can be DNA (default) or RNADNA
to make a DNA double helix or a hybrid RNA/DNA double helix.  Currently
an RNA double helix is not supported.
<p>
<b>Algorithm:</b> Basepair atomic templates (no hydrogens) are placed
along the path twisting by 36 degrees each basepair step.  Currently the
twist is not a settable parameter.  The templates are based on PDB model
1BNA.
</p>
</blockquote>

<a href="#top" class="nounder">&bull;</a>
  <a name="minimizeBackbone"><b>rna minimizeBackbone</b></a> &nbsp;<i>molecule</i>&nbsp;
  [&nbsp;<b>chunkSize</b>&nbsp;<i>c</i>&nbsp;]
  [&nbsp;<b>steps</b>&nbsp;<i>steps</i>&nbsp;]
  [&nbsp;<b>conjugateGradientSteps</b>&nbsp;<i>cgsteps</i>&nbsp;]
  [&nbsp;<b>updateInterval</b>&nbsp;<i>i</i>&nbsp;]
  [&nbsp;<b>nogui</b>&nbsp;<b>true</b>&nbsp;|&nbsp;false&nbsp;]
<br>
<blockquote>
Energy minimizes the backbone of RNA or DNA molecules keeping the
bases fixed.  The minimization is done on chunkSize (default 10)
consecutive residues at a time in order to speed up the calculation
since different parts of the backbone usually do not interact with one
another. This purpose is to fix long bonds and bad angles connecting
residues in models built by layout of nucleotide templates for example
by the rna command model or duplex operations.  Minimization is done
by the Chimera minimize command and the <b>steps</b> (default 100),
<b>conjugateGradientSteps</b> (default 100), <b>updateInterval</b>
(default 10), and <b>nogui</b> (default true) options are the same
as the minimize command nsteps, cgsteps, interval, and nogui options.
</blockquote>

<a href="#top" class="nounder">&bull;</a>
  <a name="model"><b>rna model</b></a> &nbsp;<i>sequence</i>&nbsp;
  [&nbsp;<b>path</b>&nbsp;<i>spec</i>&nbsp;]
  [&nbsp;<b>startSequence</b>&nbsp;<i>i</i>&nbsp;]
  [&nbsp;<b>pairs</b>&nbsp;<i>p</i>&nbsp;]
  [&nbsp;<b>length</b>&nbsp;<i>l</i>&nbsp;]
  [&nbsp;<b>circle</b>&nbsp;true&nbsp;|&nbsp;<b>false</b>&nbsp;]
  [&nbsp;<b>randomBranchTilt</b>&nbsp;<i>r</i>&nbsp;]
  [&nbsp;<b>loopColor</b>&nbsp;<i>color</i>&nbsp;]
  [&nbsp;<b>stemColor</b>&nbsp;<i>color</i>&nbsp;]
  [&nbsp;<b>name</b>&nbsp;<i>n</i>&nbsp;]
<br>
<blockquote>
Creates an atomic model of single-strand RNA depicting the
stems and loops with specified base pairs.  The <b>sequence</b>
argument specifies the nucleotide sequence as a string of
upper case letters AGCUT or a path to a FASTA file.  The model can
use just a subsequence starting at a position specified by the optional
<b>startSequence</b> argument (default 1).  The <b>path</b>
argument specifies a sequence of markers (e.g. #2) for the path for the
RNA to follow and those markers.  The markers must be created with the rna
path command because there is a nucleotide orientation associated with
each marker that is assigned by that command.  An alternative to specifying
the <b>path</b> argument is to specify the <b>pairs</b> argument which
is the same as described for the rna path command.  That will be used
to compute a path using the rna path command and the optional arguments
<b>length</b>, <b>circle</b>, and <b>randomBranchTilt</b>.  The atomic model
atoms and ribbon are colored according to the <b>loopColor</b> and
<b>stemColor</b> values which can be a named color or red,green,blue
or red,green,blue,opacity floating point values in the range 0 to 1.
The <b>name</b> argument specifies the new atomic model's name
(default "RNA").
<p>
<b>Algorithm:</b> The path of the RNA is either precomputed with the
rna path command or computed implicitly by the same algorith as described
for the rna path command.  That algorithm also assigns a base orientation
for each marker.  Atomic model templates for each base are placed on the
markers with the computed orientation.  The templates (file rna-templates.pdb
in the RNALayout module) place base N9 atom
of nucleotides A and G and base N1 atom of nucleotides C and U at the
path marker position (template coordinate 0,0,0).  The orientation of the
nucleotides in stems are chosen to form hydrogen bonds for each basepair
and in lobe-shaped loops they are stacked perpendicular to the plane of
the lobe and extend toward the center line of the lobe or center of the
semicircle that forms the lobe end.
</p>
</blockquote>

<a href="#top" class="nounder">&bull;</a>
  <a name="path"><b>rna path</b></a>  &nbsp;<i>pairs</i>&nbsp;
  [&nbsp;<b>length</b>&nbsp;<i>l</i>&nbsp;]
  [&nbsp;<b>circle</b>&nbsp;true&nbsp;|&nbsp;<b>false</b>&nbsp;]
  [&nbsp;<b>radius</b>&nbsp;<i>r</i>&nbsp;]
  [&nbsp;<b>randomBranchTilt</b>&nbsp;<i>r</i>&nbsp;]
  [&nbsp;<b>loopColor</b>&nbsp;<i>color</i>&nbsp;]
  [&nbsp;<b>stemColor</b>&nbsp;<i>color</i>&nbsp;]
  [&nbsp;<b>name</b>&nbsp;<i>n</i>&nbsp;]
<br>
<blockquote>
Creates a path of markers, one marker per nucleotide, depicting the
stems and loops of an RNA molecule using specified base pairs.
Consecutive markers and basepaired markers are connected by links.
The markers are the same as those made by the volume tracer tool.
The base pairs are specified as triples of integers.  For example
1,50,10 means that nucleotide 1 is basepaired to nucleotide 50, and
that is the start of a 10 base pair stem, 2 and 49, 3 and 48, ... 10 and 41
are paired.  Multiple stems are specified by giving additional triples,
such as 1,50,10,60,70,2.  The pairs argument can specify any number of stems
each represented by 3 integers.  There is a limitation that if a stem starts
at 1,50 the no other stem can pair residues in the range 1-50 with residues
outside that range.  For instance 1,50,10,20,100,5 cannot be handled by
the current layout code (may cause the code to hang or issue an error).
The stems can be specified in a 3-column text file with one line per stem
and the <b>pairs</b> argument can instead specify a file name instead of
directly specifying a comma-separated list of numbers.  Residue numbers
start at 1.  The last residue number is the highest number specified for
a stem, or that can be overridden by specifying the <b>length</b> argument.
The <b>circle</b> argument (default false) controls whether the RNA
layout is in a circle from first to last residue or a straight line.
The <b>radius</b> argument specifies the marker radius (default 2).
The <b>randomBranchTilt</b> (default 0) is an angle in degrees that causes
the code to randomly rotate stems and loops by an angle obtained by
multiplying this value by a random number uniform distributed between -1
and 1.  The <b>loopColor</b> and <b>stemColor</b> options specify the
marker colors for unpaired and paired markers and can be specified as
a color name or red,green,blue or red,green,blue,opacity floating point
values in the range 0 to 1.  The <b>name</b> argument gives the name
of the resulting marker model (default "RNA path").
<p>
<b>Algorithm:</b> The stem regions are twisted double helices rotating
36 degrees per basepair with basepairs spaced by 4 Angstroms along the
helix axis with the markers for two paired nucleotides 9 Angstroms apart
and the line between the two nucleotide markers perpendicular to the
helix axis and 4 Angstroms away from that axis.
Nucleotides that are not base paired are said to belong to loops.
Markers for loop nucleotides are 5 Angstroms apart.  The loops are laid
out in lobes which are two parallel lines of nucleotides joined by a
semi-circle of nucleotides at one end.  The semi-circle contains 8
nucleotides and the straight segments contain a maximum of 10 nucleotides
for each segment.  If a loop has more than 28 nucleotides it is drawn as
two side-by-side lobes.  The RNA between the two nucleotides at the end
of a stem forms a sequence of loops and stems and those are said to form
a cycle.  A cycle is laid out in a circle, meaning the bases of loops
and bases of stems that make up a cycle lie on a planar circle.  The
loops and stems radiating from a circular cycle extend out in the plane
of the circle unless the randomBranchTilt is non-zero in which case the
tilt out of the plane of the circle with their base still on the circle.
If the circle option if false then the top-level cycle is laid out in
a straight line instead of a circle and each stem or lobe is rotated
about the previous one by 145 degrees, in addition to the random branch tilt.
Currently all these parameters are fixed in the code.
They should be added as command arguments.
</p>
</blockquote>

<a href="#top" class="nounder">&bull;</a>
  <a name="smoothPath"><b>rna smoothPath</b></a> &nbsp;<i>path</i>&nbsp;
  [&nbsp;<b>radius</b>&nbsp;<i>r</i>&nbsp;]
  [&nbsp;<b>spacing</b>&nbsp;<i>s</i>&nbsp;]
  [&nbsp;<b>kinkInterval</b>&nbsp;<i>ki</i>&nbsp;]
  [&nbsp;<b>kinkRadius</b>&nbsp;<i>kr</i>&nbsp;]
  [&nbsp;<b>name</b>&nbsp;<i>n</i>&nbsp;]
<br>
<blockquote>
Creates a path of markers that is a smoothed version of a specified path.
The smoothing assures a minimum radius of curvature 
specified by the <b>radius</b> parameter (default 50) and produces equi-spaced
markers with spacing given by the <b>spacing</b> parameter (default 3.33).
The original use for this was to create duplex DNA path that
approximates a single strand RNA path without any sharp kinks.
A different minimum radius of curvature given by <b>kinkRadius</b>
can be applied every <b>kinkInterval</b> markers (default is no kinks).
The intent of that option is to allow some sharper kinks in the path.
The <b>name</b> argument specifies the name of the new marker model
(default "smooth path").
<p>
<b>Algorithm:</b> The first new marker is place at the first marker of the
specified path.  Subsequent markers are placed in the direction from the
most recent new marker to the next input marker.  If that direction would
cause a bend with radius of curvature less than required as judged by
the angle made by the last segment of the new path and the next segment
then that angle is expanded so the radius of curvature is the minimum value
(bend of 2*asin(spacing/2*radius)).  The next marker is placed in this
direction the specified distance away from the previous marker.
</p>
</blockquote>

<h2>
Example
</h2>

<p>
Here is an example of each of the rna operations illustrated on the first
100 nucleotides of HIV RNA with secondary structure described in
<p>

<blockquote>
Architecture and secondary structure of an entire HIV-1 RNA genome.<br>

Watts JM, Dang KK, Gorelick RJ, Leonard CW, Bess JW Jr, Swanstrom R, Burch CL, Weeks KM.<br>

Nature. 2009 Aug 6;460(7256):711-6.
PubMed <a href="http://www.ncbi.nlm.nih.gov/pubmed/19661910">19661910</a>.
</blockquote>

<p>
and for a view of the full length HIV RNA model made with this command
see this <a href="http://youtu.be/i-__ufO9LYg">HIV virus animation</a>.
</p>


<table>
<tr>
<td width=250><img src="images/hiv-path.jpg">
<td><img src="images/hiv-rna.jpg">
<td><img src="images/hiv-rna-atoms.jpg">
<td><img src="images/hiv-rna-min.jpg">
<tr>
<td align=center>rna path pairings.txt length 100
<td align=center>rna model hiv-pNL4-3.fasta #0 start 455
<td align=center>Without minimization.
<td align=center>rna min #1
<tr>
<td><img src="images/hiv-smooth.jpg">
<td><img src="images/hiv-dna.jpg">
<tr>
<td align=center>rna smooth #0
<td align=center>rna duplex #2 hiv-pNL4-3.fasta start 455
</table>

<p>
with base pairing file pairings.txt containing
<p>
<pre>
1	57	3
5	54	11
17	43	5
25	38	4
58	104	8
67	94	3
70	90	4
</pre>

<p>
and sequence file hiv-pNL4-3.fasta containing
</p>
<pre>
>gi|296556482|gb|AF324493.2| HIV-1 vector pNL4-3, 1-9709
TGGAAGGGCTAATTTGGTCCCAAAAAAGACAAGAGATCCTTGATCTGTGGATCTACCACACACAAGGCTA
CTTCCCTGATTGGCAGAACTACACACCAGGGCCAGGGATCAGATATCCACTGACCTTTGGATGGTGCTTC
AAGTTAGTACCAGTTGAACCAGAGCAAGTAGAAGAGGCCAATGAAGGAGAGAACAACAGCTTGTTACACC
CTATGAGCCAGCATGGGATGGAGGACCCGGAGGGAGAAGTATTAGTGTGGAAGTTTGACAGCCTCCTAGC
ATTTCGTCACATGGCCCGAGAGCTGCATCCGGAGTACTACAAAGACTGCTGACATCGAGCTTTCTACAAG
GGACTTTCCGCTGGGGACTTTCCAGGGAGGTGTGGCCTGGGCGGGACTGGGGAGTGGCGAGCCCTCAGAT
GCTACATATAAGCAGCTGCTTTTTGCCTGTACTGGGTCTCTCTGGTTAGACCAGATCTGAGCCTGGGAGC
TCTCTGGCTAACTAGGGAACCCACTGCTTAAGCCTCAATAAAGCTTGCCTTGAGTGCTCAAAGTAGTGTG
</pre>

</body></html>
