<html>

<!--
=== UCSF ChimeraX Copyright ===
Copyright 2016 Regents of the University of California.
All rights reserved.  This software provided pursuant to a
license agreement containing restrictions on its disclosure,
duplication and use.  For details see:
http://www.rbvi.ucsf.edu/chimerax/docs/licensing.html
This notice must be embedded in or attached to all copies,
including partial copies, of the software or any revisions
or derivations thereof.
=== UCSF ChimeraX Copyright ===
-->

<head>
<link rel="stylesheet" type="text/css" href="../userdocs.css" />
<title>Tool: AlphaFold</title>
</head><body>

<a name="top"></a>
<a href="../index.html">
<img width="60px" src="../ChimeraX-docs-icon.svg" alt="ChimeraX docs icon"
class="clRight" title="User Guide Index"/></a>

<h3><a href="../index.html#tools">Tool</a>: AlphaFold</h3>
<p>
<a href="https://alphafold.ebi.ac.uk/" target="_blank">AlphaFold</a>
is an artificial intelligence method for predicting protein structures
that has been highly successful in recent tests.
The ChimeraX <b>AlphaFold</b> tool:
<ul>
<li><a href="#search">finds and retrieves existing models</a>
from the <a href="https://alphafold.ebi.ac.uk/"
target="_blank">AlphaFold Database</a>
<li><a href="#predict">runs new AlphaFold predictions</a>
using <a href="https://colab.research.google.com/signup" 
target="_blank">Google Colab</a> and
<a href="https://github.com/deepmind/alphafold/#model-parameters-license"
target="_blank">learned parameters</a> 
</ul>
<p>
Both are freely available for academic and commercial use under
<a href="https://creativecommons.org/licenses/by/4.0/"
target="_blank">CC BY 4.0</a>.
The related tool <a href="#pae"><b>AlphaFold Error Plot</b></a>
plots residue-residue alignment errors for AlphaFold structures.
</p><p>
Users should cite:
</p>
<blockquote>
<a href="https://www.nature.com/articles/s41586-021-03819-2" target="_blank">
Highly accurate protein structure prediction with AlphaFold.</a>
Jumper J, Evans R, Pritzel A, <i>et al.</i>
<i>Nature.</i> 2021 Aug;596(7873):583-589.
</blockquote>
<a name="alphafold-multimer"></a>
<p>
...and for multimer prediction:
</p>
<blockquote>
<a href="https://www.biorxiv.org/content/10.1101/2021.10.04.463034"
target="_blank">Protein complex prediction with AlphaFold-Multimer.</a>
Evans R, O'Neill M, Pritzel A, <i>et al.</i> bioRxiv 2021.
doi: https://doi.org/10.1101/2021.10.04.463034.
</blockquote>
<p>
The
<a href="https://alphafold.ebi.ac.uk/" target="_blank">AlphaFold Database</a>
contains modeled structures for protein sequences in
<a href="https://www.uniprot.org" target="_blank">UniProt</a>:
</p>
<ul>
<li>version 1 &ndash; predictions for ~360,000 sequences,
the full reference proteomes of
<a href="https://alphafold.ebi.ac.uk/download" target="_blank">21 species</a>
including <i>Homo sapiens</i>; used by ChimeraX 1.3
<li>version 2 &ndash; predictions for ~800,000 sequences,
augmented from version 1 to include most of the manually curated entries in
<a href="https://www.uniprot.org" target="_blank">UniProt</a>;
used by ChimeraX 1.4 and newer
</ul>
<p>
...and is described in:
</p>
<blockquote>
<a href="https://academic.oup.com/nar/advance-article/doi/10.1093/nar/gkab1061/6430488" target="_blank">
AlphaFold Protein Structure Database: massively expanding the structural coverage of protein-sequence space with high-accuracy models.</a>
Varadi M, Anyango S, Deshpande M, <i>et al.</i>
<i>Nucleic Acids Res.</i> 2021 Nov 17:gkab1061. doi: 10.1093/nar/gkab1061.
</blockquote>
<p>
The predicted structures vary in
confidence levels and should be interpreted with caution.
The database contains structures for single chains, not complexes;
assembling the individual structures into a complex may give
unphysical results where parts of the chains intersect or
interact poorly with one another.
</p><p>
The <b>AlphaFold</b> tool is also implemented as the
<a href="../commands/alphafold.html"><b>alphafold</b></a> command.
See also: <a href="blastprotein.html"><b>Blast Protein</b></a>,
<a href="modeller.html"><b>Modeller Comparative</b></a>,
<a href="modelloops.html"><b>Model Loops</b></a>,
<a href="https://www.rbvi.ucsf.edu/chimerax/data/alphafold-aug2021/alphafold.html"
target="_blank">AlphaFold vs. experiment</a>,
<a href="https://www.rbvi.ucsf.edu/chimerax/data/sbgrid-mar2022/alphafold_pae.html"
target="_blank">AlphaFold for cryoEM modeling (2022)</a>
(<a href="https://www.rbvi.ucsf.edu/chimerax/data/alphafold-nov2021/af_sbgrid.html"
target="_blank">2021</a>),
<a href="https://www.rbvi.ucsf.edu/chimerax/data/alphafold-dec2021/af_multimer.html"
target="_blank">AlphaFold for complexes</a>, and
<a href="https://www.rbvi.ucsf.edu/chimerax/docs/videos/"
target="_blank">ChimeraX videos</a>:
<a href="https://youtu.be/x-9urTQ1rJY" target="_blank">searching by sequence</a>
for AlphaFold models (first 5 minutes),
<a href="https://youtu.be/W6KsBfNwDgs" target="_blank">fetching</a> an
AlphaFold model,
<a href="https://youtu.be/vj3sQfvWfDU" target="_blank">matching</a>
a known assembly structure,
<a href="https://youtu.be/le7NatFo8vI" target="_blank">predicting</a>
a structure,
<a href="https://youtu.be/6lXeCPuTePs" target="_blank">predicting
a complex</a>
</p>
<blockquote>
<a href="#dialog"><b>AlphaFold Dialog</b></a>
<br>
<a href="#coloring"><b>AlphaFold Coloring Dialog</b></a>
<br>
<a href="#pae"><b>AlphaFold Error Plot</b></a>
</blockquote>

<a name="dialog"></a>
<p class="nav">
[<a href="#top">back to top: AlphaFold</a>]
</p>
<h3>AlphaFold Dialog</h3>
<p>
The <b>AlphaFold</b> tool can be opened from the <b>Structure Prediction</b>
section of the <b>Tools</b> menu and manipulated like other panels
(<a href="../window.html#panels">more...</a>).
</p><p>
The <b>Sequence</b> can be specified by
<a href="https://www.uniprot.org/" target="_blank">UniProt</a>
name or accession number, pasted in as plain text, or chosen
from the menu of currently open protein structure chains.
</p><p>
For predicting a complex (multimer), the sequences of all chains in the complex
must be given. The same sequence must be given multiple times if it occurs in 
multiple copies in the complex. The sequences can be specified
either collectively as a model number chosen from the menu of currently
open models (<i>e.g.</i> when that model contains multiple chains), or
individually within a comma-separated list of
<a href="https://www.uniprot.org/" target="_blank">UniProt</a>
identifiers or pasted-in amino acid sequences.
Prediction may only be feasible for smaller complexes
(<a href="#caveats">details...</a>).
</p>
<ul>
<!--
When <a href="#alphafold-multimer">AlphaFold-Multimer</a> is predicting a
complex and finds multiple similar sequences from the same organism, it
needs to decide which sequences are orthologous to which chain of the target.
Sequences from the same organism are paired in order of their rank similarity 
to the target sequences.
-->
<li>
<a name="match"></a>
<b>Fetch</b> gets the most sequence-similar model available from the
<a href="https://alphafold.ebi.ac.uk/" target="_blank">AlphaFold Database</a>
for each specified chain. 
Specifying a whole model specifies all of its protein chains.
For each chain, a model is obtained for the exact UniProt entry 
if available, otherwise the single top hit identified by
<a href="https://en.wikipedia.org/wiki/BLAT_(bioinformatics)"
target="_blank">BLAT</a>-searching the 
<a href="https://alphafold.ebi.ac.uk/" target="_blank">AlphaFold Database</a>
(<a href="../commands/alphafold.html#search-option">details...</a>).
The corresponding command is
<a href="../commands/alphafold.html#match"><b>alphafold match</b></a>.
If the sequence was specified by structure chain, then:
</p>
<ol>
<li>the chain ID of the predicted structure is made the same as the
corresponding chain of the existing model
<li>the predicted structure is superimposed onto the existing chain using
<a href="../commands/matchmaker.html"><b>matchmaker</b></a>,
and the following are reported in a table
in the <a href="log.html"><b>Log</b></a>:
  <ul>
  <li><b>Chain</b> &ndash; chain ID
  <li><a href="https://www.uniprot.org/help/entry_name" 
  target="_blank"><b>UniProt Name</b></a>
  and <a href="https://www.uniprot.org/help/accession_numbers"
  target="_blank"><b>UniProt Id</b></a> (accession number)
  <li><b>RMSD</b> &ndash; C&alpha; root-mean-square deviation between
  the predicted and experimental structures, over all residues of the latter
  <li><b>Length</b> &ndash; number of residues in the predicted structure
  <li><b>Seen</b> &ndash; number of residues with atomic coordinates 
  in the experimental structure
  <li><b>% Id</b> &ndash; percent identity in the sequence alignment generated 
  by <a href="../commands/matchmaker.html"><b>matchmaker</b></a> 
  for superposition; the number of positions with identical residues
  divided by the length of the shorter sequence
  </ul>
<li>the predicted structure is trimmed to the same residue range as the 
existing chain (<a href="../commands/alphafold.html#trim">details...</a>)
<li>the following <a href="../attributes.html">attributes</a>
 are assigned to the residues of the predicted structure:
  <ul>
  <li><b>c_alpha_distance</b>
&ndash; C&alpha; distance between corresponding positions of the predicted 
and existing chains after their superposition (step 2 above) 
  <li><b>missing_structure</b>
&ndash; positions missing from the coordinates of the existing chain
  <li><b>same_sequence</b>
&ndash; positions with different residue types than the existing chain
  </ul>
These <a href="../attributes.html">attributes</a>
can be used for <a href="#coloring">coloring</a> and other purposes.
</ol>
<p>
The fetched models are stored locally in 
<b>~/Downloads/ChimeraX/AlphaFold/</b>,
where ~ indicates a user's home directory. 
If a file specified for opening is not found in this local cache,
the file will be fetched and cached. 
</p>
<li>
<a name="search"></a>
<b>Search</b>
uses a <a href="../tools/blastprotein.html#notes">BLAST</a> web service
hosted by the <a href="https://www.cgl.ucsf.edu/Overview/index.html" 
target="_blank">UCSF RBVI</a> to search the 
<a href="https://alphafold.ebi.ac.uk/" target="_blank">AlphaFold Database</a>
using default parameters: <b>BLOSUM62</b>
amino acid similarity matrix for scoring the hits,
similarity score cutoff e-value <b>1e-3</b>,
returning a maximum of <b>100</b> of unique sequences
(however, more than 100 hits may be obtained because multiple structures 
or other sequence-database entries may have the same sequence).
The corresponding command is
<a href="../commands/alphafold.html#search"><b>alphafold search</b></a>.
<b>Search</b> differs from <b>Fetch</b> in that it uses BLAST instead of BLAT,
accepts only a single chain or sequence as input,
and returns a list of hits for the user to inspect,
rather than fetching the single top hit per chain automatically.
When results are returned, the hits are listed in a
<a href="blastprotein.html#results"><b>Blast Protein</b></a> window.
Clicking the name of a hit uses
<a href="../commands/alphafold.html#fetch"><b>alphafold fetch</b></a> 
to retrieve the model.
<br><br>
<li>
<a name="predict"></a>
<b>Predict</b>
runs an <a href="https://alphafold.ebi.ac.uk/" target="_blank">AlphaFold</a>
version 2.2.0
calculation using <a href="https://colab.research.google.com/signup" 
target="_blank">Google Colab</a>, which is provided under the
<a href="https://www.apache.org/licenses/LICENSE-2.0" target="_blank">Apache
2.0</a> license. The corresponding command is
<a href="../commands/alphafold.html#predict"><b>alphafold predict</b></a>.
A warning will appear saying that this Colab notebook is from github 
(was not authored by Google), with a button to click to run anyway.
Users will need to have a Google account and to sign into it via
a browser. Once that is done, the sign-in may be remembered depending
on the user's browser settings; it is not kept in the 
<a href="../preferences.html">ChimeraX preferences</a>.
</p><p>
<a name="options"></a>
Clicking <b>Options</b> shows/hides an additional option for <b>Predict</b>:
</p>
<ul>
<a name="minimize"></a>
<li><b>Energy-minimize predicted structures</b>
&ndash; when running a new prediction, whether to minimize the result;
turning this off allows for faster job completion and/or
avoidance of failures during minimization.
</ul>
<p>
A single calculation may take hours to complete
(see the <a href="#caveats">caveats</a> below).
The free version of Colab
limits jobs to 12 hours and may terminate them at shorter times at Google's
discretion (see the <a href="https://colab.research.google.com/signup"
target="_blank">FAQ</a>).
Those who want to run longer and/or more frequent calculations
may need to sign up for one of the paid
<a href="https://colab.research.google.com/signup"
target="_blank">Colab plans</a>.
</p><p>
The model for a sequence that was specified by structure chain will
be superimposed on that chain and assigned structure-comparison attributes 
for further analysis
(<a href="../commands/alphafold.html#structmatch">details...</a>).
</p><p>
<a name="caveats"></a>
Please note the following caveats:
</p>
<ul>
<li>Prediction calculations run a <b>simplified implementation</b> of AlphaFold 
that does not use structural templates and that
searches a smaller set of protein sequences 
(<a href="https://colab.research.google.com/github/deepmind/alphafold/blob/main/notebooks/AlphaFold.ipynb" 
target="_blank">details...</a>).
The instance used by ChimeraX also limits the sequence length
and the number of sequences in the multiple sequence alignment.
For best reliability, it is recommended to use the 
<a href="https://github.com/deepmind/alphafold/" target="_blank">full 
open-source AlphaFold</a> or the <a href="https://alphafold.ebi.ac.uk/"
target="_blank">AlphaFold Database</a> (<i>e.g.</i>, with 
<a href="#search"><b>alphafold search</b></a>) instead.
<br><br>
<li>A single prediction <b>may take hours to complete</b> (see
<a href="https://www.rbvi.ucsf.edu/chimerax/data/alphafold-jan2022/afspeed.html"
target="_blank">performance statistics</a>).
The process includes installing various software packages on a virtual machine,
searching sequence databases, generating a multiple sequence alignment,
predicting atomic coordinates, and by <a href="#minimize">default</a>,
energy-minimizing the best structure.
Predicting a multimer (complex) structure may take longer than predicting
the structure of a monomer with the same total number of residues.
Those who want to run longer and/or more frequent calculations
may need to sign up for one of the paid
<a href="https://colab.research.google.com/signup"
target="_blank">Colab plans</a> for greater computational resources.
<br><br>
<li>Total <b>sequence length cannot be very large</b>.
AlphaFold runs out of graphics memory for long sequences (~800 amino acids
on old Google Colab GPUs with 16 GB memory). 
Multimer predictions face the same limit on the total number of residues,
so <b>only smaller complexes can be predicted</b>. 
As mentioned above, paid <a href="https://colab.research.google.com/signup"
target="_blank">Colab plans</a> provide more computational resources
than the free plan.
</ul>
<br><br>
<li><b>Coloring</b> 
shows the <a href="#coloring"><b>Alphafold Coloring</b> dialog</a>
for applying different color schemes to the predicted structures,
as well as hiding, showing, and 
<a href="../selection.html">selecting</b></a> their residues based on 
<a href="../attributes.html">attribute</a> value.
<br><br>
<li><b>Error plot</b> draws the <a href="#pae">AlphaFold Error Plot</a>,
in which color gradations show (for each pairwise combination of residues)
the expected error in position of one residue when the
true and predicted structures are aligned based on the other residue.
</ul>

<a name="coloring"></a>
<p class="nav">
[<a href="#top">back to top: AlphaFold</a>]
</p>
<h3>AlphaFold Coloring Dialog</h3>
<p>
Clicking the <b>Coloring</b> button on the main
<a href="#dialog"><b>AlphaFold</b> tool</a> shows the
<b>AlphaFold Coloring</b> dialog for applying different color schemes
to the predicted structures, as well as hiding, showing, and 
<a href="../selection.html">selecting</b></a> their residues based on 
<a href="../attributes.html">attribute</a> value.
</p><p>
<a name="confidence"></a>
When first opened, AlphaFold-predicted structures are automatically colored by 
the <a href="https://alphafold.ebi.ac.uk/faq#faq-5" target="_blank">pLDDT
confidence measure</a> in the B-factor field:
</p>
  <ul>
     <li>100
<table border cellpadding="4" cellspacing="0"
style="display:inline-block; vertical-align:middle">
<tr><td bgcolor=blue width="10px" title="blue"></td></tr>
</table>
to 90
<table border cellpadding="4" cellspacing="0"
style="display:inline-block; vertical-align:middle">
<tr><td bgcolor=cornflowerblue width="10px" title="cornflower blue"></td></tr>
</table>
&ndash; high accuracy expected
     <li>90
<table border cellpadding="4" cellspacing="0"
style="display:inline-block; vertical-align:middle">
<tr><td bgcolor=cornflowerblue width="10px" title="cornflower blue"></td></tr>
</table>
to 70
<table border cellpadding="4" cellspacing="0"
style="display:inline-block; vertical-align:middle">
<tr><td bgcolor=yellow width="10px" title="yellow"></td></tr>
</table>
&ndash; backbone expected to be modeled well
     <li>70
<table border cellpadding="4" cellspacing="0"
style="display:inline-block; vertical-align:middle">
<tr><td bgcolor=yellow width="10px" title="yellow"></td></tr>
</table>
to 50
<table border cellpadding="4" cellspacing="0"
style="display:inline-block; vertical-align:middle">
<tr><td bgcolor=orange width="10px" title="orange"></td></tr>
</table>
&ndash; low confidence, caution
     <li>50
<table border cellpadding="4" cellspacing="0"
style="display:inline-block; vertical-align:middle">
<tr><td bgcolor=orange width="10px" title="orange"></td></tr>
</table>
to 0
<table border cellpadding="4" cellspacing="0"
style="display:inline-block; vertical-align:middle">
<tr><td bgcolor=red width="10px" title="red"></td></tr>
</table>
&ndash; should not be interpreted, may be disordered
     </ul>
<p>
...in other words, using
</p>
<blockquote><b>
<a href="../commands/color.html#bfactor">color bfactor</a>
<a href="../commands/color.html#palette-options">palette</a> alphafold
<!-- 0,red:50,orange:70,yellow:90,cornflowerblue:100,blue -->
<!-- 100,blue:90,cornflowerblue:70,yellow:50,orange:0,red -->
</b></blockquote>
<p>
The <a href="colorkey.html"><b>Color Key</b></a> graphical interface
or a command can be used to draw a corresponding color key, for example:
</p>
  <blockquote>
  <a href="../commands/key.html"><b>key</b></a> <b>red:low orange: yellow: cornflowerblue: blue:high</b>
  &nbsp;[<a href="../commands/key.html#options"><i>other-key-options</i></a>]
<!--
<br>
  <b>key</b> <b>blue:100 cornflowerblue:90 yellow:70 orange:50 red:0</b>
  &nbsp;<b>numericLabelSpacing equal</b>
  &nbsp;[<a href="../commands/key.html#options"><i>other-key-options</i></a>]
-->
  </blockquote>
<p>
In the <b>AlphaFold Coloring</b> dialog,
the <b>Residues</b> to act on are specified by using the menus to choose an
AlphaFold-predicted model and one of the following:
</p>
  <ul>
  <li><b>all</b>
&ndash; all residues
  <li><b>confidence below</b> [<i>N</i>]
&ndash; based on the <b>bfactor</b>
<a href="../attributes.html#atom">atom attribute</a>
(the <a href="#confidence">confidence value</a> is read from the
B-factor field of the PDB file)
  <li><b>C-alpha distance greater than</b> [<i>d</i>]
&ndash; based on the <b>c_alpha_distance</b>
residue attribute of AlphaFold models 
<a href="#match">fetched by existing structure chain</a>;
C&alpha; distance between corresponding positions of the predicted and
existing chains after their automatic superposition
  <li><b>missing structure</b>
&ndash; based on the <b>missing_structure</b> 
residue attribute of AlphaFold models 
<a href="#match">fetched by existing structure chain</a>;
positions missing from the coordinates of the existing chain
  <li><b>different sequence</b>
&ndash; based on the <b>same_sequence</b> 
residue attribute of AlphaFold models 
<a href="#match">fetched by existing structure chain</a>;
positions with different residue types than the existing chain
  <li><b>confidence above</b> [<i>N</i>]
&ndash; based on the <b>bfactor</b>
<a href="../attributes.html#atom">atom attribute</a>
(the <a href="#confidence">confidence value</a> is read from the
B-factor field of the PDB file)
  <li><b>C-alpha distance less than</b> [<i>d</i>]
&ndash; based on the <b>c_alpha_distance</b>
residue attribute of AlphaFold models 
<a href="#match">fetched by existing structure chain</a>;
C&alpha; distance between corresponding positions of the predicted and
existing chains after their automatic superposition
  <li><b>paired structure</b>
&ndash; based on the <b>missing_structure</b> 
residue attribute of AlphaFold models 
<a href="#match">fetched by existing structure chain</a>;
positions present in the coordinates of the existing structure chain
  <li><b>same sequence</b>
&ndash; based on the <b>same_sequence</b> 
residue attribute of AlphaFold models 
<a href="#match">fetched by existing structure chain</a>;
positions with identical residue types as the existing chain
  </ul>
<p>
Buttons act on the designated residues:
</p>
  <ul>
  <li><b>Color</b> buttons:
    <ul>
    <li><b>Custom</b> for choosing a color interactively
    using the system color picker
    <li>a series of square buttons for specific colors:
<table cellpadding="8px" cellspacing="5x"
style="display:inline-block; vertical-align:middle"><tr>
<td bgcolor=lightgray width="1px" title="lightgray"></td>
<td bgcolor=red width="1px" title="red"></td>
<td bgcolor=magenta width="1px" title="magenta"></td>
<td bgcolor=yellow width="1px" title="yellow"></td>
<td bgcolor=lime width="1px" title="lime"></td>
<td bgcolor=cyan width="1px" title="cyan"></td>
</tr></table>
    </ul>
  <li><b>Hide</b> &ndash; hide the specified residues
  <li><b>Show</b> &ndash; show the specified residues
  <li><b>Select</b> &ndash; <a href="../selection.html">select</a>
the specified residues
  </ul>
<p>
The <b>AlphaFold Coloring</b> dialog does not color continuously 
along a gradient to show the attribute values. 
For coloring along a gradient, see the commands
<a href="../commands/color.html#bfactor"><b>color bfactor</b></a>
(for the <a href="#confidence">confidence value</a>, which is read from the
B-factor field of the PDB file) and
<a href="../commands/color.html#byattribute"><b>color byattribute</b></a>
(for other numerical <a href="../attributes.html">attributes</a>).
</p>

<a name="pae"></a>
<p class="nav">
[<a href="#top">back to top: AlphaFold</a>]
</p>
<h3>AlphaFold Error Plot</h3>
<p>
Besides the per-residue
<a href="https://alphafold.ebi.ac.uk/faq#faq-5" target="_blank">pLDDT
confidence measure</a>, AlphaFold gives
for each pair of residues (X,Y) the expected position error
at residue X if the predicted and true structures were aligned on residue Y.
These residue-residue &ldquo;predicted aligned error&rdquo; or PAE values
can be shown with <b>AlphaFold Error Plot</b>, which can be
opened from the <b>Structure Prediction</b>
section of the <b>Tools</b> menu and manipulated like other panels
(<a href="../window.html#panels">more...</a>).
It is also implemented as the command
<a href="../commands/alphafold.html#pae"><b>alphafold pae</b></a>.
</p><p>
Choosing the corresponding AlphaFold structure from the menu of open 
atomic models associates it with the plot. This association allows
coloring the structure as described below, and for selections
on the plot to highlight the corresponding parts of the structure.
</p><p>
The PAE values can be either:
</p>
<ul>
<li>fetched from the <a href="https://alphafold.ebi.ac.uk/"
target="_blank">AlphaFold Database</a>
by giving the UniProt name or accession number of an entry in that database
<br>&ndash; or &ndash;
<li>read from a json or pkl file previously output by AlphaFold
</ul>
<p>
The PAE plot is colored as follows:
</p>
<table border cellpadding="0" cellspacing="0"
style="display:inline-block; vertical-align:top"><tr>
<td bgcolor="#1e3546661e35" width="30px" title="#1e3546661e35"
align="center"><font size="-1" color="white">0</font></td>
<td bgcolor="#245a922c245a" width="30px" title="#245a922c245a"
align="center"><font size="-1" color="white">5</font></td>
<td bgcolor="#374cb168374c" width="30px" title="#374cb168374c"
align="center"><font size="-1" color="white">10</font></td>
<td bgcolor="#5687c9ba5687" width="30px" title="#5687c9ba5687"
align="center"><font size="-1">15</font></td>
<td bgcolor="#828fddf3828f" width="30px" title="#828fddf3828f"
align="center"><font size="-1">20</font></td>
<td bgcolor="#bae1efdebae1" width="30px" title="#bae1efdebae1"
align="center"><font size="-1">25</font></td>
<td bgcolor=white width="30px" title="white"
align="center"><font size="-1">30</font></td>
</tr></table>
<p>
Clicking <b>Color PAE Domains</b> clusters the residues into
coherent domains (sets of residues with relatively low PAE values)
and uses randomly chosen colors to distinguish these domains in the structure 
(<a href="../commands/alphafold.html#colorDomains">details...</a>).
Clicking <b>Color pLDDT</b> returns the structure to
the default <a href="#confidence">confidence coloring</a>.
</p><p>
The plot can also be shown by clicking the <b>Error plot</b> button
on the <a href="#dialog"><b>AlphaFold</b> dialog</a> or by using 
<a href="../commands/alphafold.html#fetch"><b>alphafold fetch</b></a>
with <b>pae true</b>.
</p>

<hr>
<address>UCSF Resource for Biocomputing, Visualization, and Informatics / 
April 2022</address>
</body></html>
