<html>

<!--
=== UCSF ChimeraX Copyright ===
Copyright 2016 Regents of the University of California.
All rights reserved.  This software provided pursuant to a
license agreement containing restrictions on its disclosure,
duplication and use.  For details see:
http://www.rbvi.ucsf.edu/chimerax/docs/licensing.html
This notice must be embedded in or attached to all copies,
including partial copies, of the software or any revisions
or derivations thereof.
=== UCSF ChimeraX Copyright ===
-->

<head>
<link rel="stylesheet" type="text/css" href="../userdocs.css" />
<title>Tool: Sequence Viewer</title>
</head><body>

<a name="top"></a>
<a href="../index.html">
<img width="60px" src="../ChimeraX-docs-icon.svg" alt="ChimeraX docs icon"
class="clRight" title="User Guide Index"/></a>

<table class="clRight">
<tr><td>
<img class="outline" src="1bzm-others-screenshot.png"
alt="1bzm, 1urt, sequence alignment" width="800px"></td></tr>
<tr><td width="790px" align="center">
<font size=-1>
Models #1 <font style="background:tan;padding:0px 2px">1bzm</font>
and #2 <font style="background:skyblue;padding:0px 2px">1urt</font>
associated with sequences in a multiple alignment, with
<font style="background:lightgreen;padding:0px 2px">ChimeraX selection</b></font>
sequence region.
</font>
</td></tr>
</table>

<h3><a href="../index.html#tools">Tool</a>: Sequence Viewer</h3>
<p>
The <b>Sequence Viewer</b> displays individual sequences and sequence alignments
of amino acids and nucleotides, with crosstalk to any
<a href="#association">associated</a> structures.
A <a href="../commands/open.html#sequence">sequence file</a> can be opened 
with the command <a href="../commands/open.html"><b>open</b></a>,
and the sequence of a structure chain can be shown
by clicking a link in the <a href="log.html"><b>Log</b></a>
(in the <b>Chain information</b> table that appears when a structure is opened)
or by using the command:
</p>
<blockquote>
<b>sequence chain 
&nbsp;<a href="../commands/atomspec.html#hierarchy"><i>chain-spec</i></a></b>
</blockquote>
<p>
Multiple <b>Sequence Viewer</b> windows can be shown at the same time, 
and they can be manipulated like other panels in ChimeraX 
(<a href="../window.html#toolstart">more...</a>).
<p>

<a name="regions"></a>
<p class="nav">
[<a href="#top">back to top</a>]
</p>
<h3>Colored Boxes (Regions)</h3>
<p>
Sequence <b><i>regions</i></b> are colored boxes or outlines
that enclose one or more residue symbols.
A single region may contain any number of disjoint and/or abutting boxes.
Pausing the cursor over a region (but not directly over a residue symbol)
shows its name in a pop-up balloon.
</p><p>
Some regions are created automatically, such as by 
<a href="#association">association</a> with a structure.
Regions can also be created manually by dragging within the sequence window.
Manually created regions are shown in
<font style="background:rgb(233,218,198);padding:1px 3px">this&nbsp;color</font>.
</p><p>
<a name="active"></a>
The <b><i>active region</i></b> is the region most recently clicked or created
by dragging.
</a>
<b>Shift</b>-dragging with the left mouse button <i>adds to</i> the active region,
whereas <b>Ctrl</b>-dragging <i>creates</i> a new region
and makes it the active region. (On Mac, command key replaces Ctrl.)
The active region is indicated with a dashed outline; clicking the active region
deactivates it, and clicking a different region deactivates the
former active region and makes the new region active.
A region with no interior color is only responsive to clicks on its borders.
<!--
Where regions <a href="#overlap">overlap</a>,
only the highest is responsive to clicks.
-->
</p>

<a name="association"></a>
<p class="nav">
[<a href="#top">back to top</a>]
</p>
<h3>Sequence-Structure Association</h3>
<p>
A sequence that arose from a structure's chain information
is automatically associated with that chain, 
with <a href="#regions">regions</a> for protein
<font style="background:rgb(255,255,204);
border:solid goldenrod 1px;padding:1px 3px">
<b>structure&nbsp;helices</b></font> and
<font style="background:lightcyan;
border:solid rgb(168,191,191) 1px;padding:1px 3px">
<b>structure&nbsp;strands</b></font>.
Such sequences are closed when their structure chains are deleted or closed,
and they are not automatically associated with other structures.
</p><p>
When sequences and structures are opened independently,
a structure chain will be associated automatically with a sequence 
if their sequences can be aligned without mismatches 
exceeding 1/10 of the residues in the structure chain.
The association and number of mismatches will be reported
in the <a href="log.html"><b>Log</b></a>. For automatic association,
<ul>
<li>Gaps in the structure sequence 
(relative to the sequence opened independently)
are not allowed except where residues are missing from the structure,
such as in disordered regions with poorly defined density.
<li>Insertions in the structure sequence relative to the 
sequence opened independently are not allowed.
</ul>
<!--
For manual (forced) association, both of the above are allowed, 
but only the &ldquo;extra&rdquo; residues in the structure not associated 
with any positions in the sequence (minus the length difference
if the structure sequence is longer) are counted along with mismatches 
and reported in the <a href="log.html"><b>Log</b></a>.
-->
</p><p>
In the sequence window, <a href="#regions">regions</a> may be created to show
<font style="background:rgb(255,76,76);padding:1px 3px">
<b>mismatches</b></font> and
<font style="border:solid black 1px;padding:1px 3px">
<b>missing&nbsp;residues</b></font>
according to the association.
If a mismatch or missing residue only occurs in a subset of multiple
associated structures, a 
<font style="background:pink;padding:1px 3px">
pink box</font> or 
<font style="border:solid rgb(190,190,190) 1px;padding:1px 3px">
gray outline</font> is shown instead.
</p><p>
Further effects of association:
</p>
<ul>
<li>a color swatch matching the associated model is shown around the 
sequence name, or a 
<font style="border:dashed darkgreen 1px;padding:1px 3px">
dark-green dashed outline</font> if chains from multiple models are associated
<li>pausing the cursor over the sequence name reports the association, and 
over residues in the sequence window reports the corresponding structure residue(s)
<li>clicking or dragging to update the <a href="#active">active region</a> will
<a href="../selection.html">select</a> the corresponding structure residues
<li><a href="../selection.html">selections</a> (by any means) of the structure
are shown as the <a href="#regions">region</a> named
<font style="background:lightgreen;padding:1px 3px">
<b>ChimeraX&nbsp;selection</b></font>
</ul>

<hr>
<address>UCSF Resource for Biocomputing, Visualization, and Informatics / 
August 2017</address>
</body></html>
